extension/storage/redisstorage/factory.go

```diff
@@ -0,0 +1,39 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//go:generate mdatagen metadata.yaml
+
```

Comment:
```suggestion
```
We already have this in `doc.go`, and I think that's a better "entrypoint" file anyway.